presto-main/src/main/java/com/facebook/presto/transaction/DelegatingTransactionManager.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.transaction;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.CatalogMetadata;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class DelegatingTransactionManager
```

Comment:
We have a discussion about whether this should be more testing oriented or it is more general.  

The argument to make it general is because it can be used later , and this pattern looks quite common in Java. So we decide keep it as is.